x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/convert/ToDateRange.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.esql.expression.function.scalar.convert;
+
+import org.elasticsearch.common.io.stream.NamedWriteableRegistry;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.xpack.esql.core.expression.Expression;
+import org.elasticsearch.xpack.esql.core.tree.NodeInfo;
+import org.elasticsearch.xpack.esql.core.tree.Source;
+import org.elasticsearch.xpack.esql.core.type.DataType;
+import org.elasticsearch.xpack.esql.expression.function.FunctionInfo;
+import org.elasticsearch.xpack.esql.expression.function.Param;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+public class ToDateRange extends AbstractConvertFunction {
+    public static final NamedWriteableRegistry.Entry ENTRY = new NamedWriteableRegistry.Entry(
+        Expression.class,
+        "ToDateRange",
+        ToDateRange::new
+    );
+
+    private static final Map<DataType, BuildFactory> EVALUATORS = Map.ofEntries(
+        // TODO
```

Comment:
OK. This looks missing!

Reply:
The obvious case would be to support strings, which would simply mean to parse the date-range using the `..` literal syntax. You could have a csv-spec test like:

```
ROW text = "2025-10-10T12:00:00..2025-10-10T12:00:00"
EVAL date_range = TO_DATE_RANGE(text)
```

This would be sufficient to test this. The implementation would also be trivial, just add one `@ConvertEvaluator` method. I could push a small commit showing this, if you want?

Reply:
I realize I can think of a compromise here, where we implement only the identity function, and leave the parse-string for a followup PR.